src/utils.c

```diff
@@ -2414,17 +2419,39 @@ void utils_start_new_geany_instance(const gchar *doc_path)
 }
 
 
+#if defined(G_OS_WIN32) && !defined(HAVE_REALPATH)
+/* realpath implementation for Windows found at http://bugzilla.gnome.org/show_bug.cgi?id=342926
+ * this one is better than e.g. liberty's lrealpath because this one uses Win32 API and works
+ * with special chars within the filename */
+static char *realpath(const char *pathname, char *resolved_path)
+{
+	int size;
+
+	if (resolved_path != NULL)
+	{
+		size = GetFullPathNameA(pathname, PATH_MAX, resolved_path, NULL);
+		if (size > PATH_MAX)
+			return NULL;
+		else
+			return resolved_path;
+	}
+	else
+	{
+		size = GetFullPathNameA(pathname, 0, NULL, NULL);
+		resolved_path = g_new0(char, size);
+		GetFullPathNameA(pathname, size, resolved_path, NULL);
+		return resolved_path;
+	}
+}
+#endif
+
+
 /**
  * Get a link-dereferenced, absolute version of a file name.
  *
  * This is similar to the POSIX `realpath` function when passed a
  * @c NULL argument.
```

Comment:
I think we can be clear here and say that this *is* realpath, except on Windows.